src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+
+      if (herbivore.health <= 0) {
+        const fatallyInjuredAnimal = Animal.alive.indexOf(herbivore);
+
+        Animal.alive.splice(fatallyInjuredAnimal, 1);
```

Comment:
The current implementation uses `indexOf` and `splice` within the `bite` method, which can be inefficient if there are many animals in the `Animal.alive` array. Consider using `filter` to create a new array without the deceased animal, which avoids the performance cost of shifting elements after removal.